src/gamepad/gamepad.vala

```diff
@@ -0,0 +1,98 @@
+// This file is part of GNOME Games. License: GPLv3
+
+/**
+ * This class represents a gamepad
+ *
+ * The client interfaces with this class primarily
+ */
+public class LibGamepad.Gamepad : Object {
+
+	/**
+	 * Emitted when a button is pressed/released
+	 * @param  button        The button pressed
+	 * @param  value         True if pressed, False if released
+	 */
+	public signal void button_event (StandardGamepadButton button, bool value);
+
+	/**
+	 * Emitted when an axis's value changes
+	 * @param  axis          The axis number from 0 to axes_number
+	 * @param  value         The value of the axis ranging from -1 to 1
+	 */
+	public signal void axis_event (StandardGamepadAxis axis, double value);
+
+	/**
+	 * Emitted when the gamepad is unplugged
+	 */
+	public signal void unplugged ();
+
+
+	/**
+	 * The name present in our database
+	 */
+	public string? name { get; private set; }
+
+	/**
+	 * The raw gamepad behind this gamepad
+	 */
+	public RawGamepad raw_gamepad { get; private set; }
```

Comment:
It's probably better to make it a private field, especially as we don't use it outside (which is good) and we shouldn't expose it.